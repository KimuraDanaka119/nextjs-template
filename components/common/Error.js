import React from 'react';
import PropTypes from 'prop-types';

import withI18N from '../hoc/withI18N';

const Error = ({ t, statusCode }) => (
  <div className="error">
    <p>
      {statusCode
        ? t('error:content.withStatusCode', { statusCode })
        : t('error:content.withoutStatusCode')
      }
    </p>
  </div>
);

Error.propTypes = {
  t: PropTypes.func,
  statusCode: PropTypes.number,
};

export default withI18N(Error, ['error']);
