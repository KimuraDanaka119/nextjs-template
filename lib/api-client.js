import fetch from 'isomorphic-unfetch';
import config from 'configs/config';

export default class ApiClient {
  async doFetch(path) {
    const url = config.api.baseURL + path;

    const res = await fetch(url);
    const data = await res.json();

    return {
      statusCode: res.status,
      data,
    };
  }

  async getShows(keyword) {
    const res = await this.doFetch(`/search/shows?q=${keyword}`);

    return res;
  }

  async getShow(id) {
    const res = await this.doFetch(`/shows/${id}`);

    return res;
  }
}
