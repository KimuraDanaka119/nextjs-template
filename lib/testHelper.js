import config from '../configs/config';

function t(key) {
  return key;
}

export { t, config };
