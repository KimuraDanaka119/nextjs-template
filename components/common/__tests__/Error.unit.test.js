import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { undecorated as Error } from '../Error';
import { t } from '../../../lib/testHelper';

describe('<Error />', () => {
  it('should render correctly with statusCode', () => {
    const error = shallow(<Error t={t} statusCode='404' />);
    expect(toJson(error)).toMatchSnapshot();
  });

  it('should render correctly without statusCode', () => {
    const error = shallow(<Error t={t} />);
    expect(toJson(error)).toMatchSnapshot();
  });
});
