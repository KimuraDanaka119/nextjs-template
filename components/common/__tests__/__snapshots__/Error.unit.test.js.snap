// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly with statusCode 1`] = `
<div
  className="error"
>
  <p>
    error:content.withStatusCode
  </p>
</div>
`;

exports[`<Error /> should render correctly without statusCode 1`] = `
<div
  className="error"
>
  <p>
    error:content.withoutStatusCode
  </p>
</div>
`;
